pkg/filetree/depth_first_path_walker.go

```diff
@@ -31,6 +31,8 @@ type WalkConditions struct {
 	// Whether we should consider children of this Node to be included in the traversal path.
 	// Return true to traverse children of this Node.
 	ShouldContinueBranch func(file.Path, filenode.FileNode) bool
+
+	LinkStrategy *LinkResolutionStrategy
```

Comment:
I don't think this is a strong enough of a case to expose the LinkResolutionStrategy struct. That is, allowing overriding of `FollowAncestorLinks` is not advised and may result in potentially odd results. Instead I feel this should use the existing API surface area:
```suggestion
	LinkOptions []LinkResolutionOption
```
... and later craft a strategy internally
```golang
...
	linkOpts := []LinkResolutionOption{followAncestorLinks}
	linkOpts = append(linkOpts, w.conditions.LinkOptions...)
	linkStrat := newLinkResolutionStrategy(linkOpts...)
...
```

Reply:
Sweet!!! I'll make some adjustments. Thanks for the review :D